components/codec/src/buffer.rs

```diff
@@ -407,6 +407,29 @@ mod tests {
     /// reallocation these values are copied.
     #[test]
     fn test_vec_reallocate() {
+        // FIXME: This test, and presumably the WriteBuffer API, relies on
+        // unspecified behavior of Vec::reserve (that it copies bytes
+        // beyond the length of the vector). It also depends on behavior
+        // specific to the malloc implementation (that calling `reserve`
+        // with a certain size _always_ reallocates).
+        //
+        // On at least one tested platform (Linux w/o jemalloc) the
+        // expected `reserve` behavior of always reallocating does not
+        // hold - malloc is free to realloc in place - so this test is
+        // "fuzzy" about exactly what it expects from the allocator;
+        // and it generates allocation "noise" to disrupt any
+        // predictive analysis in malloc.
+        //
+        // Note that the test harness for this crate uses the system
+        // allocator, not jemalloc.
+        //
+        // Both of these observations together add up to a bad smell for this
+        // API, and it might want to be reexamined. Fortunately it looks like
+        // this module is not actually used anywhere in the codebase.
```

Comment:
"this module is not actually used anywhere in the codebase" will no longer stand soon 🤣

Reply:
Yeah, I'll update this comment.